yt/visualization/line_plot.py

```diff
@@ -116,132 +122,203 @@ def __init__(self, ds, fields, start_point, end_point, npoints,
+                plot.axes.set_yscale('log')
 
-            x = x.to(unit_x)
-            y = y.to(unit_y)
+        plot._set_font_properties(self._font_properties, None)
 
-            plot.axes.plot(x, y, label=self.labels[field])
+        axes_unit_labels = self._get_axes_unit_labels(unit_x, unit_y)
 
-            if self._field_transform[field] != linear_transform:
-                if (y < 0).any():
-                    plot.axes.set_yscale('symlog')
-                else:
-                    plot.axes.set_yscale('log')
+        finfo = self.ds.field_info[field]
 
-            plot._set_font_properties(self._font_properties, None)
+        x_label = r'$\rm{Path\ Length' + axes_unit_labels[0]+'}$'
 
-            axes_unit_labels = self._get_axes_unit_labels(unit_x, unit_y)
+        finfo = self.ds.field_info[field]
+        dimensions = Unit(finfo.units,
+                          registry=self.ds.unit_registry).dimensions
+        if dimensions_counter[dimensions] > 1:
+            y_label = (r'$\rm{Multiple\ Fields}$' + r'$\rm{' +
+                       axes_unit_labels[1]+'}$')
+        else:
+            y_label = (finfo.get_latex_display_name() + r'$\rm{' +
+                       axes_unit_labels[1]+'}$')
 
-            finfo = self.ds.field_info[field]
+        plot.axes.set_xlabel(x_label)
+        plot.axes.set_ylabel(y_label)
+
+        if field in self._titles:
+            plot.axes.set_title(self._titles[field])
 
-            x_label = r'$\rm{Path\ Length' + axes_unit_labels[0]+'}$'
 
+    def _setup_plots(self):
+        if self._plot_valid is True:
+            return
+        for plot in self.plots.values():
+            plot.axes.cla()
+        dimensions_counter = defaultdict(int)
+        for field in self.fields:
             finfo = self.ds.field_info[field]
             dimensions = Unit(finfo.units,
                               registry=self.ds.unit_registry).dimensions
             dimensions_counter[dimensions] += 1
-            if dimensions_counter[dimensions] > 1:
-                y_label = (r'$\rm{Multiple\ Fields}$' + r'$\rm{' +
-                           axes_unit_labels[1]+'}$')
-            else:
-                y_label = (finfo.get_latex_display_name() + r'$\rm{' +
-                           axes_unit_labels[1]+'}$')
+        for field in self.fields:
+            plot = self._get_plot_instance(field)
 
-            plot.axes.set_xlabel(x_label)
-            plot.axes.set_ylabel(y_label)
+            x, y = self.ds.coordinates.pixelize_line(
+                field, self.start_point, self.end_point, self.npoints)
 
-            if field in self._titles:
-                plot.axes.set_title(self._titles[field])
+            self._plot_xy(field, plot, x, y, dimensions_counter,
+                          legend_label=self.labels[field])
 
             if self.include_legend[field]:
                 plot.axes.legend()
 
+
+    @invalidate_plot
+    def add_field_legend(self, field):
```

Comment:
Can this be called `annotate_legend`? That more closely follows the naming convention we use for e.g. `SlicePlot`.